sql/core/src/main/scala/org/apache/spark/sql/SQLImplicits.scala

```diff
@@ -165,11 +165,15 @@ abstract class SQLImplicits extends LowPrioritySQLImplicits {
   def newProductSeqEncoder[A <: Product : TypeTag]: Encoder[Seq[A]] = ExpressionEncoder()
 
   /** @since 2.2.0 */
-  implicit def newSequenceEncoder[T <: Seq[_] : TypeTag]: Encoder[T] = ExpressionEncoder()
+  implicit def newSequenceEncoder[T[_], E : Encoder]
```

Comment:
Can you check if we are breaking some public contract here?

Reply:
Looks like we are. I can add new methods and make the old ones not implicit. That should fix MiMa. Although that might add to the clutter that's already in this class. Is that OK?

Reply:
Just letting you know that I updated this PR with the above fix. MiMa is now passing.